pkg/app/kafka.go

```diff
@@ -0,0 +1,444 @@
+	}
+
+	log.Print("Count that we received from application is.", field.M{"app": kc.name, "count": count})
+	return count, nil
+}
+
+func (kc *KafkaCluster) Reset(ctx context.Context) error {
+	log.Print("Resetting the kafka topic is being handled in blueprint.", field.M{"app": kc.name})
+	return nil
+}
+
+// Initialize is used to initialize the database or create schema
+func (kc *KafkaCluster) Initialize(ctx context.Context) error {
+	return nil
+}
+
+// to produce messages to kafka
+func produce(ctx context.Context, topic string, namespace string) error {
+	partition := 0
+	forwarder, err := K8SServicePortForward(ctx, "my-cluster-kafka-external-bootstrap", namespace, "9094")
+	if err != nil {
+		return err
+	}
+
+	defer forwarder.Close()
+	ports, err := forwarder.GetPorts()
+	if err != nil {
+		return err
+	}
+	uri := "localhost:" + fmt.Sprint(ports[0].Local)
+
+	conn, err := kafka.DialLeader(ctx, "tcp", uri, topic, partition)
+	if err != nil {
+		return err
+	}
+	err = conn.SetWriteDeadline(time.Now().Add(10 * time.Second))
+	if err != nil {
+		return err
+	}
+	_, err = conn.WriteMessages(
+		kafka.Message{Value: []byte("{'title':'The Matrix','year':1999,'cast':['Keanu Reeves','Laurence Fishburne','Carrie-Anne Moss','Hugo Weaving','Joe Pantoliano'],'genres':['Science Fiction']}")},
+	)
+	if err != nil {
+		return err
+	}
+
+	if err := conn.Close(); err != nil {
+		return err
+	}
+	return nil
+}
+
+// LoadConfig returns a kubernetes client config based on global settings.
+func LoadConfig() (*rest.Config, error) {
+	// log.Print("Attempting to use InCluster config")
+	config, err := rest.InClusterConfig()
+	if err == nil {
+		return config, nil
+	}
+
+	homeConfig := filepath.Join(os.Getenv("HOME"), ".kube/config")
+	config, err = clientcmd.BuildConfigFromFlags("", homeConfig)
+	if err == nil {
+		return config, nil
+	}
+	return nil, errors.Wrapf(err, "Failed to load default k8s config")
+}
+
+// K8SServicePortForward creates a service port forwarding and returns the forwarder and the error if any
+func K8SServicePortForward(ctx context.Context, svcName string, ns string, pPort string) (*portforward.PortForwarder, error) {
+	var selector string
+	errCh := make(chan error)
+	readyChan := make(chan struct{}, 1)
+
+	cfg, err := LoadConfig()
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to Load config")
+	}
+
+	clientset, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to create Clienset for k8s config")
+	}
+	roundTripper, upgrader, err := spdy.RoundTripperFor(cfg)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to create RoundTripper for k8s config")
+	}
+
+	svc, err := clientset.CoreV1().Services(ns).Get(ctx, svcName, metav1.GetOptions{})
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to get service for component")
+	}
+	for k, v := range svc.Spec.Selector {
+		selector = fmt.Sprintf("%s%s=%s,", selector, k, v)
```

Comment:
Can we use https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#FormatLabelSelector?